### PR TITLE
Initial Implementation

### LINKED ISSUE #2: Initial implementation
Tech stack: NodeJS & Cassandra

Functions:
- Save files in Cassandra.  Create a NodeJS side function that takes 1. file content 2. file extension .  It should save the file in Cassandra with a unique randomly generated name.
- Serve the saved files publicly with NodeJS.

### PR DESCRIPTION
**Pull request by arkaray**
fixes #2

### CODE REVIEW

**Comment by arkaray** [comment]
There is a bit of work left

File: main.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+}
+
+function createFileTableIfNotExist() {
+  return client.execute(`CREATE TABLE IF NOT EXISTS ${SCHEMA_NAME}.${FILES_TABLE}(content BLOB, extension VARCHAR, name VARCHAR PRIMARY KEY);`);
+}
+
+const crypto = require('crypto');
+
+function generateName() {
+  const buf = crypto.randomBytes(64);
```
Comment (kkimdev) [inline main.js]: Use https://www.npmjs.com/package/uuid

**Comment by arkaray** [comment]
ptal

File: serve.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+}
+
+function createFileTableIfNotExist() {
+  return client.execute(`CREATE TABLE IF NOT EXISTS ${SCHEMA_NAME}.${FILES_TABLE}(content BLOB, extension VARCHAR, name VARCHAR PRIMARY KEY);`);
+}
+
+(async () => {
+  try {
+    await createKeyspaceIfNotExist();
+    await createFileTableIfNotExist();
```
Comment (kkimdev) [inline serve.js]: Creating table should be in writing part.  There is no case that serving needs to create one.  If there is no table, nothing to serve

File: serve.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+}
+
+app.get('/*', async (req, res) => {
+  const { pathname } = url.parse(req.url);
+  try {
+    file = await getFile(pathname);
+    res.attachment(pathname);
+    res.write(file);
+    log.info(`File Served: ${pathname}`);
+  } catch (e) {
```
Comment (kkimdev) [inline serve.js]: this block is handling 404 error, then catch that error specifically, not everything.

For other errors, log the error and return 500.  Btw, client requesting not-existing file is not a server error, don't log it.

File: write.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+  function _generateName() {
+    return uuidv4();
+  }
+
+  async function write(content, extension) {
+    const fileName = this._generateName();
+    this.client.execute(`INSERT INTO ${SCHEMA_NAME}.${FILES_TABLE}(content,extension,name)VALUES(?,?,?)`, [content, extension, fileName]);
+    return fileName;
+  }
```
Comment (kkimdev) [inline write.js]: missing error handling

File: serve.js (hunk 3 above)
Comment (arkaray) [reply serve.js]: return 500 for all cases. postponing the handling of non-existent files.

File: serve.js (hunk 2 above)
Comment (arkaray) [reply serve.js]: done

File: write.js (hunk 4 above)
Comment (arkaray) [reply write.js]: error handling in app.get.

**Comment by arkaray** [comment]
ptal

File: serve.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    },
+  });
+
+const PORT = argv.port;
+const CASSANDRA_HOST = argv.dbhost;
+const SCHEMA_NAME = 'schema2';
+const FILES_TABLE = 'files';
+
+const log = bunyan.createLogger({
+  name: 'main.js',
```
Comment (kkimdev) [inline serve.js]: incorrect name?

**Comment by kkimdev** [comment]
Please continue working on this branch for other changes.

**Pull request merged**